### PR TITLE
Add "root" crate for TypeSpec

### PR DESCRIPTION
**Pull request by heaths**
Serves as a package-as-a-namespace for RFC 3243:
https://rust-lang.github.io/rfcs/3243-packages-as-optional-namespaces.html.

We may put common types between possible client- and service-side code
generators in here as well.

### CODE REVIEW

**Comment by heaths** [comment]
If approved, I plan to publish this immediately.

File: sdk/typespec/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "typespec"
+version = "0.1.0"
+authors.workspace = true
+edition.workspace = true
+description = "Project root for all TypeSpec-related crates"
+homepage = "https://aka.ms/typespec"
```
Comment (timotheeguerin) [inline sdk/typespec/Cargo.toml:7]:
```suggestion
homepage = "https://typespec.io"
```

Maybe that aka.ms link should be updated but it links to the telemetry app.

Comment (heaths) [reply sdk/typespec/Cargo.toml]: Good idea!

**Comment by heaths** [comment]
Adding @ronniegeraghty. We can fill out this README more (which will also appear on crates.io) as we do more with it.

**Review by LarryOsterman — APPROVED** [review]

**Pull request merged**